In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [3]:
birds = pd.read_csv("dataset\\penguins.csv")

In [4]:
birds

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1> Palmer Penguin Data Analytics </h1>
<hr>
<h3> Analyst: Christian Hardinan </h3>

<p>Import data set to Jupyter notebook</p>
<p>Convert dataset to dataframe</p>
<ul><strong>Requirements</strong>
    <li>Name of the dataframe </li>
    <li>Source</li>
</ul>

In [23]:
birds = pd.read_csv('dataset\\penguins.csv')

In [24]:
birds

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [25]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [26]:
birds.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [27]:
birds.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [28]:
birds.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>Selecting a column</h1>
<p><strong>Syntax</strong></p>

<p>data_frame['column_name']</p>

In [29]:
birds['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [30]:
birds['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [31]:
birds['species'] == "Chinstrap"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [32]:
birds['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [33]:
birds['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [34]:
chinstrap_df = birds[birds['species'] == 'Chinstrap']

In [35]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [36]:
adelie_df = birds[birds['species'] == 'Adelie']

In [37]:
gentoo_df = birds[birds['species'] == 'Gentoo']

In [38]:
birds[birds['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [39]:
birds[(birds['species'] == "Chinstrap") & (birds['body_mass_g'] >= 400)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            68 non-null     object 
 1   island             68 non-null     object 
 2   bill_length_mm     68 non-null     float64
 3   bill_depth_mm      68 non-null     float64
 4   flipper_length_mm  68 non-null     float64
 5   body_mass_g        68 non-null     float64
 6   sex                68 non-null     object 
 7   year               68 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.8+ KB


<h1> HOW TO FILLUP NULL VALUES</h1>

In [40]:
birds.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [41]:
birds['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [42]:
birds['bill_length_mm'].fillna(birds['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [43]:
birds['bill_length_mm'] = birds['bill_length_mm'].fillna(birds['bill_length_mm'].mean())
birds['flipper_length_mm'] = birds['flipper_length_mm'].fillna(birds['flipper_length_mm'].mean())
birds['body_mass_g'] = birds['body_mass_g'].fillna(birds['body_mass_g'].mean())

In [44]:
birds.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [45]:
birds['sex'] = birds['sex'].fillna(birds['sex'].mode()[0])

In [46]:
birds.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [47]:
birds['sex'].mode()

0    male
Name: sex, dtype: object

<h1>HOW TO CHANGE THE DATA TYPE OF A COLUMN</h1>

In [48]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [49]:
birds['species'] = birds['species'].astype('category')

In [50]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    int64   
dtypes: category(1), float64(4), int64(1), object(2)
memory usage: 19.4+ KB


In [51]:
birds.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     342.000000         344.000000   344.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.443643       1.974793          14.020657   799.613058   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.250000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [52]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    int64   
dtypes: category(1), float64(4), int64(1), object(2)
memory usage: 19.4+ KB


In [53]:
birds.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [54]:
birds['mass_kg'] = np.round(birds['body_mass_g'] / 1000, 1)

In [55]:
birds

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7         181.000000   
1       Adelie  Torgersen        39.50000           17.4         186.000000   
2       Adelie  Torgersen        40.30000           18.0         195.000000   
3       Adelie  Torgersen        43.92193            NaN         200.915205   
4       Adelie  Torgersen        36.70000           19.3         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8         207.000000   
340  Chinstrap      Dream        43.50000           18.1         202.000000   
341  Chinstrap      Dream        49.60000           18.2         193.000000   
342  Chinstrap      Dream        50.80000           19.0         210.000000   
343  Chinstrap      Dream        50.20000           18.7         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1>QUESTIONS</h1>
<h2> 1. Which species has the longest bill length? </h2>

In [56]:
birds

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7         181.000000   
1       Adelie  Torgersen        39.50000           17.4         186.000000   
2       Adelie  Torgersen        40.30000           18.0         195.000000   
3       Adelie  Torgersen        43.92193            NaN         200.915205   
4       Adelie  Torgersen        36.70000           19.3         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000           19.8         207.000000   
340  Chinstrap      Dream        43.50000           18.1         202.000000   
341  Chinstrap      Dream        49.60000           18.2         193.000000   
342  Chinstrap      Dream        50.80000           19.0         210.000000   
343  Chinstrap      Dream        50.20000           18.7         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [57]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    int64   
 8   mass_kg            344 non-null    float64 
dtypes: category(1), float64(5), int64(1), object(2)
memory usage: 22.1+ KB


In [58]:
birds['bill_length_mm'].value_counts()

bill_length_mm
41.1    7
45.2    6
37.8    5
39.6    5
50.0    5
       ..
51.0    1
53.5    1
51.4    1
51.9    1
55.8    1
Name: count, Length: 165, dtype: int64

In [59]:
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

In [60]:
adelie_df['bill_length_mm'].mean()

np.float64(38.79139072847682)

In [61]:
gentoo_df['bill_length_mm'].mean()

np.float64(47.50487804878048)

<h1>INSIGHT#1</h1>
<p><strong>The data shows the mean of each species:</strong></p>
<p> chinstrap - 48.83382352941177</p>
<p> adelie - 38.82514427516159</p>
<p> gentoo - 47.475983305036785</p>
<p><strong>in which the chinstrap with the total of 48.83382352941177, has the longest bill length average</strong></p>

<h1> 2. Which island has the most number of male birds? </h1>

In [118]:
birds['island'].unique()

array(['Torgersen', 'Biscoe', 'Dream'], dtype=object)

In [119]:
torgersen_df = birds[birds['island'] == 'Torgersen']

In [120]:
torgersen_male = torgersen_df['sex'] == 'male'

In [65]:
torgersen_male.value_counts()

sex
True     28
False    24
Name: count, dtype: int64

In [66]:
biscoe_df = birds[birds['island'] == 'Biscoe']

In [67]:
biscoe_male = biscoe_df['sex'] == 'male'

In [68]:
biscoe_male.value_counts()

sex
True     88
False    80
Name: count, dtype: int64

In [69]:
dream_df = birds[birds['island'] == 'Dream']

In [70]:
dream_male = biscoe_df['sex'] == 'male'

In [71]:
dream_male.value_counts()

sex
True     88
False    80
Name: count, dtype: int64

<h1>INSIGHT#2 </h1>
<p><strong>The data shows the numbers of male in every island</strong></p>
<p> Torgersen = 28</p>
<p> Biscoe = 88</p>
<p> Dream = 88</p>
<p><strong>The results shows that both Biscoe and Dream with the average of 88 male, have the moast numbers of male birds</strong></p>

<h1> 3. Which island has the most number of female birds </h1>

In [72]:
birds['island'].unique()

array(['Torgersen', 'Biscoe', 'Dream'], dtype=object)

In [73]:
torgersen_df = birds[birds['island'] == 'Torgersen']

In [74]:
torgersen_female = torgersen_df['sex'] == 'female'

In [75]:
torgersen_female.value_counts()

sex
False    28
True     24
Name: count, dtype: int64

In [76]:
biscoe_df = birds[birds['island'] == 'Biscoe']

In [77]:
biscoe_female = biscoe_df['sex'] == 'female'

In [78]:
biscoe_female.value_counts()

sex
False    88
True     80
Name: count, dtype: int64

In [79]:
dream_df = birds[birds['island'] == 'Dream']

In [80]:
dream_female = biscoe_df['sex'] == 'female'

In [81]:
dream_male.value_counts()

sex
True     88
False    80
Name: count, dtype: int64

<h1>INSIGHT#3 </h1>
<p><strong>The data shows the numbers of female in every island</strong></p>
<p> Torgersen = 24</p>
<p> Biscoe = 80</p>
<p> Dream = 80</p>
<p><strong>The results shows that both Biscoe and Dream with the average of 80 female, have the moast numbers of female birds</strong></p>

<h1> 4. What is the average bill length of Adelie? </h1>

In [82]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [83]:
adelie_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 152 entries, 0 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            152 non-null    object 
 1   island             152 non-null    object 
 2   bill_length_mm     151 non-null    float64
 3   bill_depth_mm      151 non-null    float64
 4   flipper_length_mm  151 non-null    float64
 5   body_mass_g        151 non-null    float64
 6   sex                146 non-null    object 
 7   year               152 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 10.7+ KB


In [84]:
adelie_df['bill_length_mm'].value_counts()

bill_length_mm
41.1    7
37.8    5
39.6    5
39.7    4
40.6    4
       ..
43.1    1
36.8    1
35.6    1
32.1    1
40.7    1
Name: count, Length: 78, dtype: int64

In [85]:
adelie_df['bill_length_mm'].mean()

np.float64(38.79139072847682)

<h1>INSIGHT#4</h1>
<p><strong>The mean of Adelie is 38.79139072847682</strong></p>

<h1> 5. What is the average bill length of Chinstrap? </h1>

In [86]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [87]:
chinstrap_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            68 non-null     object 
 1   island             68 non-null     object 
 2   bill_length_mm     68 non-null     float64
 3   bill_depth_mm      68 non-null     float64
 4   flipper_length_mm  68 non-null     float64
 5   body_mass_g        68 non-null     float64
 6   sex                68 non-null     object 
 7   year               68 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.8+ KB


In [88]:
chinstrap_df['bill_length_mm'].value_counts()

bill_length_mm
51.3    3
52.0    3
45.2    2
50.5    2
46.4    2
50.8    2
42.5    2
49.0    2
50.9    2
45.7    2
50.2    2
46.1    1
52.7    1
45.9    1
46.6    1
50.3    1
58.0    1
45.4    1
46.0    1
47.0    1
51.7    1
50.0    1
46.5    1
46.7    1
50.6    1
43.2    1
48.5    1
42.4    1
49.2    1
54.2    1
49.5    1
47.5    1
49.7    1
51.0    1
47.6    1
53.5    1
46.9    1
40.9    1
52.8    1
45.5    1
46.2    1
49.8    1
50.1    1
48.1    1
51.4    1
50.7    1
51.5    1
52.2    1
49.3    1
45.6    1
51.9    1
46.8    1
55.8    1
43.5    1
49.6    1
Name: count, dtype: int64

In [89]:
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

<h1>INSIGHT#5</h1>
<p><strong>The mean of Chinstrap is 48.83382352941177</strong></p>

<h1> 6. What is the average bill length of Gentoo? </h1>

In [90]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [91]:
gentoo_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 152 to 275
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            124 non-null    object 
 1   island             124 non-null    object 
 2   bill_length_mm     123 non-null    float64
 3   bill_depth_mm      123 non-null    float64
 4   flipper_length_mm  123 non-null    float64
 5   body_mass_g        123 non-null    float64
 6   sex                119 non-null    object 
 7   year               124 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 8.7+ KB


In [92]:
gentoo_df['bill_length_mm'].value_counts()

bill_length_mm
50.0    4
46.5    4
45.5    4
46.2    4
45.2    4
       ..
48.1    1
51.5    1
55.1    1
48.8    1
49.9    1
Name: count, Length: 75, dtype: int64

In [93]:
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

<h1>INSIGHT#6</h1>
<p><strong>The mean of Gentoo is 48.83382352941177</strong></p>

<h1> 7. What is the total number of male and female of Gentoo? </h1>

In [94]:
gentoo_df['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

<h1>INSIGHT#7</h1>
<h2><strong>The total number of male and female from Gentoo species is 119</strong></h2>

<h1>8. What is the average bill depth of adelie species?</h1>

In [107]:
adelie_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 152 entries, 0 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            152 non-null    object 
 1   island             152 non-null    object 
 2   bill_length_mm     151 non-null    float64
 3   bill_depth_mm      151 non-null    float64
 4   flipper_length_mm  151 non-null    float64
 5   body_mass_g        151 non-null    float64
 6   sex                146 non-null    object 
 7   year               152 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 10.7+ KB


In [108]:
adelie_df['bill_depth_mm'].value_counts

bill_depth_mm
18.5    9
18.6    8
18.9    8
17.0    7
18.1    7
17.1    6
17.2    6
17.9    6
19.0    5
17.8    5
18.0    5
20.0    4
19.1    4
17.6    4
18.4    4
17.5    4
18.8    4
21.1    3
18.7    3
19.2    3
18.3    3
19.5    3
19.4    3
17.3    2
19.3    2
16.1    2
16.8    2
16.5    2
16.6    2
20.7    2
21.2    2
17.7    2
16.9    2
20.3    2
17.4    1
19.6    1
20.2    1
20.6    1
19.7    1
21.5    1
16.7    1
16.2    1
18.2    1
16.0    1
20.5    1
19.8    1
15.9    1
20.1    1
15.5    1
Name: count, dtype: int64

In [109]:
adelie_df['bill_depth_mm'].mean()

np.float64(18.346357615894043)

<h1>INSIGHT#8</h1>
<h2><strong>The average bill depth of Adelie species is 18.346357615894043</strong></h2>

<h1>9. What is the average bill depth of <strong>biscoe island<strong>?</h1>

In [113]:
biscoe_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 168 entries, 20 to 275
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            168 non-null    category
 1   island             168 non-null    object  
 2   bill_length_mm     168 non-null    float64 
 3   bill_depth_mm      167 non-null    float64 
 4   flipper_length_mm  168 non-null    float64 
 5   body_mass_g        168 non-null    float64 
 6   sex                168 non-null    object  
 7   year               168 non-null    int64   
 8   mass_kg            168 non-null    float64 
dtypes: category(1), float64(5), int64(1), object(2)
memory usage: 12.1+ KB


In [114]:
biscoe_df['bill_depth_mm'].value_counts

<bound method IndexOpsMixin.value_counts of 20     18.3
21     18.7
22     19.2
23     18.1
24     17.2
       ... 
271     NaN
272    14.3
273    15.7
274    14.8
275    16.1
Name: bill_depth_mm, Length: 168, dtype: float64>

In [117]:
biscoe_df['bill_depth_mm'].mean()

np.float64(15.8748502994012)

<h1>INSIGHT#9</h1>
<h2><strong>The highest average bill depth of Biscoe island  is 15.8748502994012</strong></h2>

<h1>10. What species have the most number in <strong>biscoe  island<strong>?</h1>

In [147]:
biscoe_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20   Adelie  Biscoe        37.80000           18.3         174.000000   
21   Adelie  Biscoe        37.70000           18.7         180.000000   
22   Adelie  Biscoe        35.90000           19.2         189.000000   
23   Adelie  Biscoe        38.20000           18.1         185.000000   
24   Adelie  Biscoe        38.80000           17.2         180.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe        43.92193            NaN         200.915205   
272  Gentoo  Biscoe        46.80000           14.3         215.000000   
273  Gentoo  Biscoe        50.40000           15.7         222.000000   
274  Gentoo  Biscoe        45.20000           14.8         212.000000   
275  Gentoo  Biscoe        49.90000           16.1         213.000000   

     body_mass_g     sex  year  mass_kg  
20   3400.000000  female  2007      3.4  
21   3600.000000    male  2007      3.6  
22   3800.000000  female  2007      3.8  
23   3950.000000    male  2007      4.0  
24   3800.000000    male  2007      3.8  
..           ...     ...   ...      ...  
271  4201.754386    male  2009      4.2  
272  4850.000000  female  2009      4.8  
273  5750.000000    male  2009      5.8  
274  5200.000000  female  2009      5.2  
275  5400.000000    male  2009      5.4  

[168 rows x 9 columns]

In [148]:
biscoe_df['species'].value_counts()

species
Gentoo       124
Adelie        44
Chinstrap      0
Name: count, dtype: int64

<h1>INSIGHT#9</h1>
<h2><strong>The data shows that Gentoo with the average of 124 species has the highest numbers of species in BISCOE ISLAND</strong></h2>

<h1>11. What species have the most number in <strong>Torgersen Island<strong>?</h1>

In [151]:
torgersen_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen        39.10000           18.7         181.000000   
1    Adelie  Torgersen        39.50000           17.4         186.000000   
2    Adelie  Torgersen        40.30000           18.0         195.000000   
3    Adelie  Torgersen        43.92193            NaN         200.915205   
4    Adelie  Torgersen        36.70000           19.3         193.000000   
5    Adelie  Torgersen        39.30000           20.6         190.000000   
6    Adelie  Torgersen        38.90000           17.8         181.000000   
7    Adelie  Torgersen        39.20000           19.6         195.000000   
8    Adelie  Torgersen        34.10000           18.1         193.000000   
9    Adelie  Torgersen        42.00000           20.2         190.000000   
10   Adelie  Torgersen        37.80000           17.1         186.000000   
11   Adelie  Torgersen        37.80000           17.3         180.000000   
12   Adelie  Torgersen        41.10000           17.6         182.000000   
13   Adelie  Torgersen        38.60000           21.2         191.000000   
14   Adelie  Torgersen        34.60000           21.1         198.000000   
15   Adelie  Torgersen        36.60000           17.8         185.000000   
16   Adelie  Torgersen        38.70000           19.0         195.000000   
17   Adelie  Torgersen        42.50000           20.7         197.000000   
18   Adelie  Torgersen        34.40000           18.4         184.000000   
19   Adelie  Torgersen        46.00000           21.5         194.000000   
68   Adelie  Torgersen        35.90000           16.6         190.000000   
69   Adelie  Torgersen        41.80000           19.4         198.000000   
70   Adelie  Torgersen        33.50000           19.0         190.000000   
71   Adelie  Torgersen        39.70000           18.4         190.000000   
72   Adelie  Torgersen        39.60000           17.2         196.000000   
73   Adelie  Torgersen        45.80000           18.9         197.000000   
74   Adelie  Torgersen        35.50000           17.5         190.000000   
75   Adelie  Torgersen        42.80000           18.5         195.000000   
76   Adelie  Torgersen        40.90000           16.8         191.000000   
77   Adelie  Torgersen        37.20000           19.4         184.000000   
78   Adelie  Torgersen        36.20000           16.1         187.000000   
79   Adelie  Torgersen        42.10000           19.1         195.000000   
80   Adelie  Torgersen        34.60000           17.2         189.000000   
81   Adelie  Torgersen        42.90000           17.6         196.000000   
82   Adelie  Torgersen        36.70000           18.8         187.000000   
83   Adelie  Torgersen        35.10000           19.4         193.000000   
116  Adelie  Torgersen        38.60000           17.0         188.000000   
117  Adelie  Torgersen        37.30000           20.5         199.000000   
118  Adelie  Torgersen        35.70000           17.0         189.000000   
119  Adelie  Torgersen        41.10000           18.6         189.000000   
120  Adelie  Torgersen        36.20000           17.2         187.000000   
121  Adelie  Torgersen        37.70000           19.8         198.000000   
122  Adelie  Torgersen        40.20000           17.0         176.000000   
123  Adelie  Torgersen        41.40000           18.5         202.000000   
124  Adelie  Torgersen        35.20000           15.9         186.000000   
125  Adelie  Torgersen        40.60000           19.0         199.000000   
126  Adelie  Torgersen        38.80000           17.6         191.000000   
127  Adelie  Torgersen        41.50000           18.3         195.000000   
128  Adelie  Torgersen        39.00000           17.1         191.000000   
129  Adelie  Torgersen        44.10000           18.0         210.000000   
130  Adelie  Torgersen        38.50000           17.9         190.000000   
131  Adelie  Torgersen        43.10000           19.

In [152]:
torgersen_df['species'].value_counts()

species
Adelie       52
Chinstrap     0
Gentoo        0
Name: count, dtype: int64

<h1>INSIGHT#11</h1>
<h2><strong>The data shows that Adelie with the average of 52 species has the highest numbers of species in TORGERSEN ISLAND</strong></h2>

<h1>12. What species have the most number in <strong>Dream Island<strong>?</h1>

In [155]:
dream_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
30        3250.0  female  2007      3.2  
31        3900.0    male  2007      3.9  
32        3300.0  female  2007      3.3  
33        3900.0    male  2007      3.9  
34        3325.0  female  2007      3.3  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[124 rows x 9 columns]

In [156]:
dream_df['species'].value_counts()

species
Chinstrap    68
Adelie       56
Gentoo        0
Name: count, dtype: int64

<h1>INSIGHT#12</h1>
<h2><strong>The data shows that CHinstrap with the average of 68 species has the highest numbers of species in DREAM ISLAND</strong></h2>

<h1>DATA AGGREGATION</h1>

In [95]:
PerIsland = birds.groupby(['species','island']).count()

In [96]:
PerIsland

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             51                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            123                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

In [97]:
birds_insight8 = birds.groupby(['species','island','sex','year']).size()

In [98]:
birds_insight8

species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Biscoe     female  2007     0
                              2008     0
                              2009     0
                      male    2007     0
                              2008     0
                      

In [99]:
avg_species_island = birds.groupby(['species','island'])['body_mass_g'].mean()

In [100]:
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Biscoe               NaN
           Dream        3733.088235
           Torgersen            NaN
Gentoo     Biscoe       5068.965761
           Dream                NaN
           Torgersen            NaN
Name: body_mass_g, dtype: float64

<h1>TESTING CORRELATION</h1>

<h3>Pesrdon R Correlation</h3>
<hr>
<p>Used for testing correlation between two <strong> NUMERICAL VALUES</strong></p>

In [104]:
weight_flipper_cor = birds['body_mass_g'].corr(birds['flipper_length_mm'])

In [105]:
weight_flipper_cor

np.float64(0.8712017673060115)